daemon/mgr/container_logger.go

```diff
@@ -5,27 +5,46 @@ import (
 
 	"github.com/alibaba/pouch/apis/types"
 	"github.com/alibaba/pouch/daemon/logger"
+	"github.com/alibaba/pouch/daemon/logger/bufferlog"
 	"github.com/alibaba/pouch/daemon/logger/jsonfile"
 	"github.com/alibaba/pouch/daemon/logger/syslog"
 
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	// LogNonBlocking means to use buffer to make logs non blocking.
+	LogNonBlocking = "non-blocking"
+)
+
 func logOptionsForContainerio(c *Container, info logger.Info) (logger.LogDriver, error) {
+	var ld logger.LogDriver
+	var err error
+
 	cfg := c.HostConfig.LogConfig
 	if cfg == nil || cfg.LogDriver == types.LogConfigLogDriverNone {
 		return nil, nil
 	}
 
 	switch cfg.LogDriver {
 	case types.LogConfigLogDriverJSONFile:
-		return jsonfile.Init(info)
+		ld, err = jsonfile.Init(info)
 	case types.LogConfigLogDriverSyslog:
-		return syslog.Init(info)
+		ld, err = syslog.Init(info)
 	default:
 		logrus.Warnf("not support (%v) log driver yet", cfg.LogDriver)
 		return nil, nil
 	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if info.LogConfig["mode"] == LogNonBlocking {
```

Comment:
what here need validate logic for this log Opt? since the mode can be `blocking` or anything else.

Reply:
Yes, in default, there is no buffer that will result in blocking if  there is a lot of data.